### PR TITLE
[WIP] Sugary

### PR DESCRIPTION
**Pull request by gj**

### CODE REVIEW

File: backends/rails/app/policy/authorization.polar [hunk 1]
```diff
@@ -1,67 +1,83 @@
...
+  "read" if "reader" on "parent";
+}
 
-actor_has_role_for_resource(actor: User, role_name, resource: Repo) if
-    role in actor.repo_roles and
-    role_name = role.name and
-    resource = role.repo;
+has_relation(repo, "parent", issue: Issue) if
+  repo = issue.repo and
+  repo matches Repo;
```
Comment (gkaemmer) [inline backends/rails/app/policy/authorization.polar:83]: Could this be a specializer?

Comment (gj) [reply backends/rails/app/policy/authorization.polar:83]: Nope; [the grounding issue](https://www.notion.so/osohq/Ground-constrained-variable-when-unifying-with-external-instance-c2f5f5054487487ebf787a595cecdc10) has reared its ugly head again

Comment (gkaemmer) [reply backends/rails/app/policy/authorization.polar:83]: 🙀

Comment (gj) [reply backends/rails/app/policy/authorization.polar]: JK- the grounding issue seems like it only applies to my original formulation of
```ruby
has_relation(repo: Repo, "parent", _: Issue{repo: repo});
```
where the unification takes place in the head. Was able to move the `matches` back into the head as long as the unification stays in the body:
```ruby
has_relation(repo: Repo, "parent", issue: Issue) if repo = issue.repo;
```

**Comment by gw000** [comment]
done in https://github.com/osohq/gitclub/pull/16

**Pull request closed without merging**